src/leiningen/hiera.clj

```diff
@@ -132,13 +137,18 @@
         context (merge default-options
                        (:hiera project)
                        {:internal-ns (set (file-namespaces source-files))
-                        :graph (file-deps source-files)})]
-    (rhizome/save-graph
-      (graph-nodes context)
-      (partial adjacent-to context)
-      :vertical? (:vertical context)
-      :node->descriptor (partial render-node context)
-      :node->cluster (partial node-cluster context)
-      :cluster->descriptor (fn [c] {:label c})
-      :filename (:path context))
-    (println "Generated namespace graph in" (:path context))))
+                        :graph (file-deps source-files)})
+        conf {:vertical? (:vertical context)
+              :node->descriptor (partial render-node context)
+              :node->cluster (partial node-cluster context)
+              :cluster->descriptor (fn [c] {:label c})}]
+    (apply (partial rhizome/save-graph
+                    (graph-nodes context)
+                    (partial adjacent-to context))
+           (flatten (vec (assoc conf :filename (:path context)))))
+    (println "Generated namespace graph in" (:path context))
+    (save-dot (apply (partial rdot/graph->dot
+                              (graph-nodes context)
+                              (partial adjacent-to context))
+                     (flatten (vec conf)))
+              (str/replace (:path context) #".png" ".dot"))))
```

Comment:
This regex is risky since the filename might contain characters matching this in other locations. As written, this will definitely cause unexpected behavior in some cases:

```clojure
=> (str/replace "/home/pnguyen/foo/target/ns-hierarchy.png" #".png" ".dot")
"/home.dotuyen/foo/target/ns-hierarchy.dot"
```

Part of the problem is that you didn't escape the `.`, making it a match-any-character pattern. You should also add a `$` at the end to anchor the match to the end of the input string.